orca-front50/src/main/java/com/netflix/spinnaker/orca/front50/tasks/SavePipelineTask.java

```diff
@@ -161,14 +193,16 @@ private void updateServiceAccount(Map<String, Object> pipeline, String serviceAc
   }
 
   private Map<String, Object> fetchExistingPipeline(Map<String, Object> newPipeline) {
-    String applicationName = (String) newPipeline.get("application");
     String newPipelineID = (String) newPipeline.get("id");
-    if (!StringUtils.isEmpty(newPipelineID)) {
-      return front50Service.getPipelines(applicationName).stream()
-          .filter(m -> m.containsKey("id"))
-          .filter(m -> m.get("id").equals(newPipelineID))
-          .findFirst()
-          .orElse(null);
+    if (StringUtils.isNotEmpty(newPipelineID)) {
+      try {
+        return front50Service.getPipeline(newPipelineID);
+      } catch (RetrofitError e) {
```

Comment:
This needs to change to 
```suggestion
      } catch (SpinnakerHttpException e) {
        // Return a null if pipeline with expected id not found
        if (e.getResponseCode() == HTTP_NOT_FOUND) {
```
since the [Front50Service bean](https://github.com/spinnaker/orca/blob/4c9d51398054f5aa0ddcd715c6204b24ca4933f6/orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/config/Front50Configuration.groovy#L82) uses
```
.setErrorHandler(SpinnakerRetrofitErrorHandler.getInstance())
```

Reply:
https://github.com/spinnaker/orca/pull/4773/commits/bbf991637a54a6f8b6d01565d96981f508cc30fc